src/Cart.js

```diff
@@ -19,15 +50,28 @@ class Cart extends Component {
                     <h1>Here are all the products in your cart!</h1>
                     <ul>
                         {lineItems.map(p=>{
+                            const total = p.quantity * p.price;
                             return (
+                                <div>
                                 <li key={p.id}>
                                     <span>{`Name: ${p.name}, Price: ${p.price}`}</span>
                                     <br/>
                                     <img className = 'product-image' src={p.imageUrl}/>
+                                    <span>{`Quantity: ${p.quantity}, Total: ${total}`}</span>
+                                    <form onSubmit={onUpdate}>
+                                        <label htmlFor='quantity'>Quantity</label>
+                                        <input name='quantity' onChange={(e) => onChange(p, e)}/>
```

Comment:
I see you are displaying the quantity, then have a separate input field to change the qty. These should probably be the same field.

Display the input field where you show the item quantity, and give it the value of the current item quantity. The user can then just change that field.

Reply:
To aid in this, you would probably want to populate the state when the component mounts, with a line items array containing objects with qty and price.

Reply:
@5t4mp5 I updated the part where the input filed show the item quantity. I set the current lineitem on state when the user is about to change it instead of having an array so that I can just send the state to the update thunk. :)